packages/toast/src/ToastProgressBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { cx, css, keyframes } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+
+const progressBackgroundBase = '#22B7EB';
+const toastWidth = 400;
+
+const progressBarBackgroundStyle = css`
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  height: 6px;
+  background-color: ${uiColors.gray.light2};
+  border-radius: 0 0 4px 4px;
+  overflow: hidden;
+`;
+
+const backgroundShimmer = keyframes`
+  0% {
+    background-position: ${-toastWidth}px;
+  }
+
+  100% {
+    background-position: ${toastWidth * 2}px;
+  }
+`;
+
+const progressBarStyle = css`
+  overflow: hidden;
+  height: 6px;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  background-color: ${progressBackgroundBase};
+  background-image: linear-gradient(
+    90deg,
+    ${progressBackgroundBase} 0px,
+    #cce8f4 ${toastWidth / 2}px,
+    ${progressBackgroundBase} ${toastWidth}px
+  );
+  background-size: 600px;
+  animation: ${backgroundShimmer} 4s infinite linear;
+  transition: width 0.3s ease-in-out;
+`;
+
+interface ProgressBarProps {
+  progress: number;
+}
+
+function ToastProgressBar({ progress }: ProgressBarProps) {
```

Comment:
can we clamp `progress` to be between 0 and 1?

Reply:
Sure! It shouldn't make a difference in the display of the component, but could make the component slightly more resilient in case we accidentally break the CSS.